test_quality_of_service/test/qos_utilities.cpp

```diff
@@ -0,0 +1,158 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <string>
+#include <tuple>
+
+#include "test_quality_of_service/qos_utilities.hpp"
+
+
+std::tuple<size_t, size_t> chrono_milliseconds_to_size_t(
```

Comment:
By the, aren't you trying to re-invent `time_t` by chance? If yes, using the type directly could be better?

Reply:
> This function name is not great, it's not clear what means size_t there and what is this tuple. It would be probably easier for the reader to read:

Naming inspired by (https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/TimeUnit.html#toMillis(long)) as it's a simple conversion method.

> By the, aren't you trying to re-invent `time_t` by chance? If yes, using the type directly could be better?

Not trying to re-invent types: most of the time usage I've seen in rclcpp using chrono, whereas the QoS settings are using ctime. Without a function the conversion work was repeated, so this was simply to illustrate how a user could set the QoS time appropriately.

Will change the method name from _chrono_milliseconds_to_size_t_ to _convert_chrono_milliseconds_to_size_t_.